app/models/invoice.rb

```diff
@@ -5,4 +5,24 @@ class Invoice < ApplicationRecord
   has_many :transactions
 
   validates :status, presence: true
+
+  def self.get_by_key(key, params)
```

Comment:
Let's discuss tomorrow. Can you articulate to me the advantages/disadvantages of doing this?